hbase-server/src/main/java/org/apache/hadoop/hbase/master/HbckChore.java

```diff
@@ -271,12 +297,12 @@ private void loadRegionsFromFS() throws IOException {
           continue;
         }
         HbckRegionInfo hri = regionInfoMap.get(encodedRegionName);
-        if (hri == null) {
+        // If it is not in in-memory database and not a merged region,
+        // report it as an orphan region.
+        if (hri == null && !mergedParentRegions.contains(encodedRegionName)) {
```

Comment:
if it's not in in-memory database but it is in merged regions, that seems like a problem as well and should be reported?

Reply:
Are you referring to the following case

`if (hri == null && mergedParentRegions.contains(encodedRegionName)) {}
`
This is valid for merged parent regions as of today.

Reply:
Oh, merge parents are NOT in in-memory state because they are not active. We just have this background cleaner task that is doing janitorial work on hbase:meta cleaning up meta and filesystem....